odoo/addons/base/models/ir_rule.py

```diff
@@ -192,19 +192,22 @@ def _make_access_error(self, operation, records):
 
         model = records._name
         description = self.env['ir.model']._get(model).name or model
-        msg_heads = {
-            # Messages are declared in extenso so they are properly exported in translation terms
-            'read':   _("Due to security restrictions, you are not allowed to access '%(document_kind)s' (%(document_model)s) records.", document_kind=description, document_model=model),
-            'write':  _("Due to security restrictions, you are not allowed to modify '%(document_kind)s' (%(document_model)s) records.", document_kind=description, document_model=model),
-            'create': _("Due to security restrictions, you are not allowed to create '%(document_kind)s' (%(document_model)s) records.", document_kind=description, document_model=model),
-            'unlink': _("Due to security restrictions, you are not allowed to delete '%(document_kind)s' (%(document_model)s) records.", document_kind=description, document_model=model)
+        operations = {
+            'read':  _("read"),
+            'write': _("write"),
+            'create': _("create"),
+            'unlink': _("unlink"),
         }
-        operation_error = msg_heads[operation]
-        resolution_info = _("Contact your administrator to request access if necessary.")
+        user_description = f"{self.env.user.name} (id={self.env.user.id})"
+        operation_error = _("Uh-oh! Looks like you have stumbled upon some top-secret records.\n\n" \
+            "Sorry my friend, %s doesn't have '%s' access to:", user_description, operations[operation])
```

Comment:
'my friend' seems a bit too friendly to me, if I may. Some cultures do not say "bro" or "friend" every two words.